src/test/java/com/redhat/rhjmc/containerjfr/platform/openshift/OpenShiftPlatformClientTest.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    @Test
+    void shouldReturnListOfMatchingEndpointRefs() throws Exception {
+        String namespace = "foo-namespace";
+        setMockNamespace(namespace);
+
+        MixedOperation mockNamespaceOperation = Mockito.mock(MixedOperation.class);
+        Mockito.when(osClient.endpoints()).thenReturn(mockNamespaceOperation);
+
+        ArgumentCaptor<String> namespaceCaptor = ArgumentCaptor.forClass(String.class);
+        NonNamespaceOperation mockOperation = Mockito.mock(NonNamespaceOperation.class);
+        Mockito.when(mockNamespaceOperation.inNamespace(namespaceCaptor.capture()))
+                .thenReturn(mockOperation);
+
+        EndpointsList mockListable = Mockito.mock(EndpointsList.class);
+        Mockito.when(mockOperation.list()).thenReturn(mockListable);
+
+        ObjectReference objRef1 = Mockito.mock(ObjectReference.class);
+        // Mockito.when(objRef1.getName()).thenReturn("targetA");
+        ObjectReference objRef2 = Mockito.mock(ObjectReference.class);
+        Mockito.when(objRef2.getName()).thenReturn("targetB");
+        ObjectReference objRef3 = Mockito.mock(ObjectReference.class);
+        Mockito.when(objRef3.getName()).thenReturn("targetC");
+        ObjectReference objRef4 = Mockito.mock(ObjectReference.class);
+        Mockito.when(objRef4.getName()).thenReturn("targetD");
+
+        EndpointAddress address1 = Mockito.mock(EndpointAddress.class);
+        // Mockito.when(address1.getIp()).thenReturn("127.0.0.1");
+        // Mockito.when(address1.getTargetRef()).thenReturn(objRef1);
+        EndpointAddress address2 = Mockito.mock(EndpointAddress.class);
+        Mockito.when(address2.getIp()).thenReturn("127.0.0.2");
+        Mockito.when(address2.getTargetRef()).thenReturn(objRef2);
+        EndpointAddress address3 = Mockito.mock(EndpointAddress.class);
+        Mockito.when(address3.getIp()).thenReturn("127.0.0.3");
+        Mockito.when(address3.getTargetRef()).thenReturn(objRef3);
+        EndpointAddress address4 = Mockito.mock(EndpointAddress.class);
+        Mockito.when(address4.getIp()).thenReturn("127.0.0.4");
+        Mockito.when(address4.getTargetRef()).thenReturn(objRef4);
+
+        EndpointPort port1 = Mockito.mock(EndpointPort.class);
+        Mockito.when(port1.getPort()).thenReturn(80);
+        Mockito.when(port1.getName()).thenReturn("tcp-80");
+        EndpointPort port2 = Mockito.mock(EndpointPort.class);
+        Mockito.when(port2.getPort()).thenReturn(9999);
+        Mockito.when(port2.getName()).thenReturn("jfr-jmx");
+        EndpointPort port3 = Mockito.mock(EndpointPort.class);
+        Mockito.when(port3.getPort()).thenReturn(9091);
+        Mockito.when(port3.getName()).thenReturn("tcp-9091");
+
+        EndpointSubset subset1 = Mockito.mock(EndpointSubset.class);
+        // Mockito.when(subset1.getAddresses()).thenReturn(Collections.singletonList(address1));
+        Mockito.when(subset1.getPorts()).thenReturn(Collections.singletonList(port1));
```

Comment:
Are these commented-out lines meant to be kept?

Reply:
Sort of. I set up all that structure of a mock endpoints API response, but because of the way the selection algorithm works, the IP and TargetRef of the Address are never checked if the corresponding port isn't deemed compatible. This caused Mockito to complain about unnecessary stubbing, although if those lines were completely removed then it might be a little more difficult to come back to that test later and see how things were intended to be structured when certain fields from the mock response are just omitted entirely.